baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/IllegalSafeLoggingArgument.java

```diff
@@ -427,4 +432,86 @@ public Description matchClass(ClassTree tree, VisitorState state) {
                         "Dangerous type: annotated '%s' but ancestors declare '%s'.", directSafety, ancestorSafety))
                 .build();
     }
+
+    @Override
+    public Description matchLambdaExpression(LambdaExpressionTree tree, VisitorState state) {
+        TreePath path = state.getPath();
+        while (path != null && !(path.getLeaf() instanceof VariableTree)) {
+            path = path.getParentPath();
+        }
+        if (path == null) {
+            // The lambda isn't directly assigned to a variable, so we don't know how to analyze it (yet)
+            // TODO(aldexis): handle other ways that a lambda can be used, e.g. returned from a method
+            return Description.NO_MATCH;
+        }
+        VariableTree variable = (VariableTree) path.getLeaf();
+        Type variableType = ASTHelpers.getType(variable);
```

Comment:
I think we can simplify the implementation and solve your TODO thusly (worth verifying with some additional test cases where a lambda is returned, or provided as an argument into a method invocation)
```suggestion
        Type variableType = ASTHelpers.getType(tree);
```

Reply:
Indeed, this worked much better! I've used this and added the relevant test cases (lamdba as return type, and lambda as argument in a method invocation)

Thanks for pointing me to this!

I was pattern-matching from the `matchReturn` method. Is the reason this doesn't work there because we want not only the return type of the method, but also the potential annotation on the method itself, which we can't derive from `ASTHelper.getType(tree)`? (I'll check with the debugger, since I'm curious, but asking in case you know/remember off the top of your head)